alacritty/src/renderer/box_drawing.rs

```diff
@@ -0,0 +1,584 @@
+                '\u{2594}' => (height / 8., height),
+                _ => (height, height),
+            };
+            // Fixup `y` coordinates.
+            let y = height - y;
+
+            let x = match character {
+                '\u{2590}' => canvas.center_h(),
+                '\u{2595}' => width as f32 - width / 8.,
+                _ => 0.,
+            };
+
+            canvas.draw_rect(x, y, rect_width, rect_height, COLOR_FILL);
+        },
+        // Shades: '░', '▒', '▓', '█'.
+        '\u{2588}' | '\u{2591}' | '\u{2592}' | '\u{2593}' => {
+            let color = match character {
+                '\u{2588}' => COLOR_FILL,
+                '\u{2591}' => COLOR_FILL_ALPHA_STEP_1,
+                '\u{2592}' => COLOR_FILL_ALPHA_STEP_2,
+                '\u{2593}' => COLOR_FILL_ALPHA_STEP_3,
+                _ => unreachable!(),
+            };
+            canvas.fill(color);
+        },
+        // Quadrants: '▖', '▗', '▘', '▙', '▚', '▛', '▜', '▝', '▞', '▟'.
+        '\u{2596}'..='\u{259F}' => {
+            let (w_second, h_second) = match character {
+                '\u{2598}' | '\u{2599}' | '\u{259a}' | '\u{259b}' | '\u{259c}' => {
+                    (canvas.center_h(), canvas.center_v())
+                },
+                _ => (0., 0.),
+            };
+            let (w_first, h_first) = match character {
+                '\u{259b}' | '\u{259c}' | '\u{259d}' | '\u{259e}' | '\u{259f}' => {
+                    (canvas.center_h(), canvas.center_v())
+                },
+                _ => (0., 0.),
+            };
+            let (w_third, h_third) = match character {
+                '\u{2596}' | '\u{2599}' | '\u{259b}' | '\u{259e}' | '\u{259f}' => {
+                    (canvas.center_h(), canvas.center_v())
+                },
+                _ => (0., 0.),
+            };
+            let (w_fourth, h_fourth) = match character {
+                '\u{2597}' | '\u{2599}' | '\u{259a}' | '\u{259c}' | '\u{259f}' => {
+                    (canvas.center_h(), canvas.center_v())
+                },
+                _ => (0., 0.),
+            };
+
+            // Second quadrant.
+            canvas.draw_rect(0., 0., w_second, h_second, COLOR_FILL);
+            // First quadrant.
+            canvas.draw_rect(canvas.center_h(), 0., w_first, h_first, COLOR_FILL);
+            // Third quadrant.
+            canvas.draw_rect(0., canvas.center_v(), w_third, h_third, COLOR_FILL);
+            // Fourth quadrant.
+            canvas.draw_rect(canvas.center_h(), canvas.center_v(), w_fourth, h_fourth, COLOR_FILL);
+        },
+        _ => unreachable!(),
+    }
+
+    let top = height as i32 + metrics.descent as i32;
+    let buffer = BitmapBuffer::Rgb(canvas.into_raw_buffer());
+    RasterizedGlyph { character, top, left: 0, height: height as i32, width: width as i32, buffer }
+}
+
+#[repr(packed)]
+#[derive(Clone, Copy, Debug, Default)]
+struct RGBPixel {
+    _r: u8,
+    _g: u8,
+    _b: u8,
+}
```

Comment:
Why the `repr(packed)` and `_` prefix here? It should also be `RgbPixel` and I think `Pixel` could potentially be omitted completely.

Reply:
packed is just to ensure that everything is packed regardless of what compiler could think here. I don't trust compilers when it comes to layouts. Like it should be byte aligned, but...

Reply:
Ordering might also be an issue, because Rust can freely reorder fileds. So repr is good here.